src/plugins/data_views/common/data_views/data_view.ts

```diff
@@ -312,6 +135,16 @@ export class DataView implements DataViewBase {
         ? this.fields.toSpec({ getFormatterForField: this.getFormatterForField.bind(this) })
         : undefined;
 
+    const fieldAttrs = cloneDeep(this.fieldAttrs);
+    if (!includeFields) {
+      Object.keys(fieldAttrs).forEach((key) => {
+        delete fieldAttrs[key].count;
+        if (Object.keys(fieldAttrs[key]).length === 0) {
+          delete fieldAttrs[key];
+        }
+      });
+    }
+
```

Comment:
Could this be breaking? CI tests don't seem to care, but it looks like it could change the output in some situations. Maybe it's low impact enough that it doesn't matter though, I can't really think of it being an issue apart from potentially if we're comparing to a previously stored spec somewhere.

Reply:
I need to review this. I suspect its a mistake.

Reply:
I was wrong - had entirely forgotten writing the code until I saw the failed tests.

Added a comment to explain. Its an idiosyncrasy that we might as well preserve. A lot of what this PR does is to separate out modifying the field list from modifying the data structures used to create the field list. Inconsistencies in field update lifecycle can result in this sort of unexpected behavior.